core/core-impl/src/main/java/ai/timefold/solver/core/impl/score/stream/bi/ConsecutiveSequencesBiConstraintCollector.java

```diff
@@ -0,0 +1,40 @@
+package ai.timefold.solver.core.impl.score.stream.bi;
+
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Supplier;
+import java.util.function.ToIntFunction;
+
+import ai.timefold.solver.core.api.score.stream.ConstraintCollectors.SequenceChain;
+import ai.timefold.solver.core.impl.score.stream.SequenceCalculator;
+
+final class ConsecutiveSequencesBiConstraintCollector<A, B, Result_>
+        extends
+        ObjectCalculatorBiCollector<A, B, Result_, SequenceChain<Result_, Integer>, SequenceCalculator<Result_>> {
+
+    private final ToIntFunction<Result_> indexMap;
+
+    public ConsecutiveSequencesBiConstraintCollector(BiFunction<A, B, Result_> resultMap, ToIntFunction<Result_> indexMap) {
+        super(resultMap);
+        this.indexMap = Objects.requireNonNull(indexMap);
+    }
+
+    @Override
+    public Supplier<SequenceCalculator<Result_>> supplier() {
+        return () -> new SequenceCalculator<>(indexMap);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof ConsecutiveSequencesBiConstraintCollector<?, ?, ?> other) {
```

Comment:
You can call `super.equals()` to check if mappers are the same

Reply:
This is intentional.

I believe that equals/hashCode are supposed to be simple and easy to understand. One shouldn't have to go to superclasses to understand what equals will do; it should be obvious.

If the superclass had many fields with complicated relationships, I'd understand wanting to reuse that code. In that case, this aspect would IMO win over readability. But in this case, we're really talking about just one field in the superclass, and that's IMO not enough to obfuscate these simple methods.